app/signin/page.tsx

```diff
@@ -0,0 +1,62 @@
+import React from "react";
+import { redirect } from "next/navigation";
+import { auth } from "@/auth";
+import Image from "next/image";
+import formImage from "../../public/likehome_form_icon.png";
+import OAuthButton from "@/components/OAuthButton";
+
+
+const loginPageURL = "/profiletest";
+
+export default async function SignInPage() {
+  const profilePageURL = "/";
+
+  // redirect user
+  const session = await auth();
```

Comment:
This could be a context provider.
Same as earlier, can you make this an issue for future dev.
Highly recommend looking into later if you have time.